metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFastTableService.java

```diff
@@ -121,6 +167,10 @@ public HiveConnectorFastTableService(
                 .query(conn, queryBuilder.toString(), handler, params.toArray());
         } catch (SQLException e) {
             throw Throwables.propagate(e);
+        } finally {
+            final long duration = registry.clock().monotonicTime() - start;
+            log.debug("### Time taken to complete getTableNames is {} ms", duration);
+            this.registry.timer(requestTimerId.withTags(tags)).record(duration, TimeUnit.MILLISECONDS);
```

Comment:
Yeah I think we should definitely use it in the metacat-main services or API tier. Can be a separate thing. We can file issue.